docs/src/cookbooks/verilog-vs-chisel.md

```diff
@@ -0,0 +1,920 @@
+    val in = Input(UInt(width.W))
+    val out = Output(UInt(width.W))
+  })
+  io.out := io.in
+}
+```
+```scala mdoc:invisible
+ChiselStage.emitVerilog(new PassthroughGenerator(10))
+```
+</td>
+         </tr>
+         <tr>
+<td>
+
+```verilog
+module ParameterizedWidthAdder(
+  input [in0Width-1:0] in0,
+  input [in1Width-1:0] in1,
+  output [sumWidth-1:0] sum
+);
+  parameter in0Width = 8;
+  parameter in1Width = 1;
+  parameter sumWidth = 9;
+
+  assign sum = in0 + in1;
+
+endmodule
+```
+
+</td>
+<td>
+
+```scala mdoc:silent
+class ParameterizedWidthAdder(
+  in0Width: Int,
+  in1Width: Int,
+  sumWidth: Int) extends Module {
+  val io = IO(new Bundle {
+    val in0 = Input(UInt(in0Width.W))
+    val in1 = Input(UInt(in1Width.W))
+    val sum = Output(UInt(sumWidth.W))
+  })
+  // a +& b includes the carry, a + b does not
+  io.sum := io.in0 +& io.in1
+}
+```
+</td>
+</tr>
+<tr>
+<td>
+
+</td>
+         </tr>
+    </table>
+<html>
+<body>
+
+# Wire Assignment and Literal Values
+
+<html>
+<body>
+    <table border ="0">
+          <tr>
+            <td><b style="font-size:30px">Verilog</b></td>
+            <td><b style="font-size:30px">Chisel</b></td>
+         </tr>
+         <tr>
+<td>
+
+```verilog
+wire [31:0] 
+a = 32'd42; 
+wire [31:0] 
+b = 32'hbabecafe; 
+wire [15:0] c; 
+assign c = 16'b1;
+```
+
+</td>
+<td>
+
+```scala mdoc:silent
+
+
+class MyWireAssignmentModule extends Module {
+ val a = WireDefault(42.U(32.W))
+ val b = WireDefault("hbabecafe".U(32.W))  
+ val c = Wire(UInt(16.W)) 
+  c := "b1".U;
+}
+```
+```scala mdoc:invisible
+ChiselStage.emitVerilog(new MyWireAssignmentModule)
+```
+</td>
+         </tr>
+<tr>
+<td>
+
+```verilog
+module MyWireAssignmentModule2 ();
+
+wire [31:0] aa = 'd42;
+ reg [31:0] a; // Logical register for use in always block, not real register
+ 
+ //
+always @(*) begin
+  a = aa;
+end 
+
+// Hex value initialization
+ wire [31:0] b = 32'hbabecafe;
+
+ // Declaration separate from Assignment
+ wire [15:0] c;
+ wire d;
+
+ assign c = 16'b1;
+ assign d = 1'b1;
+
+// Signed values
+wire signed [63:0] g;
+assign g = -’d5;
+
+wire signed [31:0] h = 'd5;
+
+reg signed[31:0] f;
+always@(*) begin
+  f = ‘d5;
+end
+endmodule
+```
+
+
+</td>
+<td>
+
+```scala mdoc:silent
+
+
+class MyWireAssignmentModule2 extends Module {
+
+val aa = 42.U(32.W)
+val a = Wire(UInt(32.W))
+a := aa
+val b = "hbabecafe".U(32.W)
+val c = Wire(UInt(16.W))
+val d = Wire(Bool())
+c := "b1".U(16.W)
+d := true.B
+val g = Wire(SInt(64.W))
+g := -5.S
+val h = 5.asSInt(32.W)
+val f = Wire(SInt(32.W))
+f := 5.S
+}
+```
+```scala mdoc:invisible
+ChiselStage.emitVerilog(new MyWireAssignmentModule2)
+```
+
+</td>
+</tr>
+    </table>
+<html>
+<body>
+
+# Register Declaration and Assignment
+
+<html>
+<body>
+    <table border ="0">
+          <tr>
+            <td><b style="font-size:30px">Verilog</b></td>
+            <td><b style="font-size:30px">Chisel</b></td>
+         </tr>
+         <tr>
+<td>
+
+```verilog
+module RegisterModule(
```

Comment:
Can we delete this first simple example `RegisterModule`, given that we have the second more complicated example below? At any rate this one is not a good example since it does not actually have a register in it.